x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRCS1FailureStoreRestIT.java

```diff
@@ -343,97 +557,40 @@ private void testResolveRemoteClustersIsUnauthorized() {
         // user with only read_failure_store access should not be able to resolve remote clusters
         var exc = expectThrows(
             ResponseException.class,
-            () -> performRequestWithUser(ONLY_READ_FAILURE_STORE_ACCESS, new Request("GET", "/_resolve/cluster/" + REMOTE_CLUSTER_ALIAS))
+            () -> performRequestMaybeUsingApiKey(ONLY_READ_FAILURE_STORE_ACCESS, new Request("GET", "/_resolve/cluster"))
         );
         assertThat(exc.getResponse().getStatusLine().getStatusCode(), equalTo(403));
         assertThat(
             exc.getMessage(),
-            containsString(
-                "action ["
-                    + "indices:admin/resolve/cluster"
-                    + "] is unauthorized for user ["
-                    + ONLY_READ_FAILURE_STORE_ACCESS
-                    + "] "
-                    + "with effective roles ["
-                    + ONLY_READ_FAILURE_STORE_ACCESS
-                    + "]"
+            anyOf(
+                containsString(
+                    "action ["
+                        + "indices:admin/resolve/cluster"
+                        + "] is unauthorized for user ["
+                        + ONLY_READ_FAILURE_STORE_ACCESS
+                        + "] "
+                        + "with effective roles ["
+                        + ONLY_READ_FAILURE_STORE_ACCESS
+                        + "]"
+                ),
+                containsString(
+                    "action [indices:admin/resolve/cluster] is unauthorized for API key id ["
+                        + apiKeys.get(ONLY_READ_FAILURE_STORE_ACCESS).v1()
+                        + "] of user ["
+                        + ONLY_READ_FAILURE_STORE_ACCESS
+                        + "]"
+                )
             )
         );
     }
 
-    private static void setupLocalDataOnQueryCluster() throws IOException {
-        // Index some documents, to use them in a mixed-cluster search
-        final var indexDocRequest = new Request("POST", "/local_index/_doc?refresh=true");
-        indexDocRequest.setJsonEntity("{\"local_foo\": \"local_bar\"}");
-        assertOK(client().performRequest(indexDocRequest));
-    }
-
-    private static void setupUserAndRoleOnQueryCluster() throws IOException {
```

Comment:
These changes are mostly refactorings in order to reuse the role definitions when creating API keys.